compiler/rustc_trait_selection/src/lib.rs

```diff
@@ -21,6 +21,8 @@
 #![feature(if_let_guard)]
 #![feature(never_type)]
 #![recursion_limit = "512"] // For rustdoc
+// #![deny(rustc::untranslatable_diagnostic)]
+// #![deny(rustc::diagnostic_outside_of_impl)]
```

Comment:
How much is left to uncomment these lines?

Reply:
These files in `rustc_trait_selection/src/traits` still have `untranslatable_diagnostic` and `diagnostic_outside_of_impl` errors 

`traits/coherence.rs` (2 errors)
`traits/const_evaluatable.rs` (4 errors)
`traits/object_safety.rs` (2 errors)
`traits/util.rs` (6 errors)

`traits/error_reporting/suggestions.rs` (146 errors)
`traits/error_reporting/mod.rs` (34 errors)

`traits/select/candidate_assembly.rs` (2 errors)
`traits/select/mod.rs` (2 errors)

`traits/specialize/mod.rs` (2 errors)

Reply:
There's still `untranslatable_diagnostic/diagnostic_outside_of_impl` errors in these files ->
```
compiler/rustc_trait_selection/src/traits/select/mod.rs (1 errors)
compiler/rustc_trait_selection/src/traits/util.rs (3 errors)
compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs (74 errors)
compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs (28 errors)
compiler/rustc_trait_selection/src/traits/select/candidate_assembly.rs (1 errors)
compiler/rustc_trait_selection/src/traits/object_safety.rs (1 errors)
compiler/rustc_trait_selection/src/traits/coherence.rs (1 errors)
compiler/rustc_trait_selection/src/traits/specialize/mod.rs (2 errors)
compiler/rustc_trait_selection/src/traits/on_unimplemented.rs (2 errors)
compiler/rustc_trait_selection/src/traits/const_evaluatable.rs (2 errors)
```

Reply:
There's still `untranslatable_diagnostic/diagnostic_outside_of_impl` errors in these files
```
compiler/rustc_trait_selection/src/traits/select/mod.rs (1 error)
compiler/rustc_trait_selection/src/traits/util.rs (3 errors)
compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs (74 errors)
compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs (28 errors)
compiler/rustc_trait_selection/src/traits/select/candidate_assembly.rs (1 error)
compiler/rustc_trait_selection/src/traits/object_safety.rs (1 error)
compiler/rustc_trait_selection/src/traits/coherence.rs (1 error)
compiler/rustc_trait_selection/src/traits/specialize/mod.rs (2 errors)
compiler/rustc_trait_selection/src/traits/on_unimplemented.rs (2 errors)
compiler/rustc_trait_selection/src/traits/const_evaluatable.rs (2 errors)
```

Reply:
`rustc_trait_selection` is a big crate, feel free to just land part of it at a time (e.g. just the modules you've done so far, or a few more, whatever works for you).

Reply:
Yeah, it is a pretty big crate! I cleaned up the changes a bit to land this patch.

Reply:
Yeah, it is a pretty big crate! I cleaned up the changes a bit to land these changes.

Reply:
Yeah, it is a pretty big crate! I cleaned up the changes a bit to land the modules I've done so far.